src/main/java/org/reaktivity/nukleus/tls/internal/stream/ServerStreamFactory.java

```diff
@@ -46,73 +52,80 @@
 
 import org.agrona.DirectBuffer;
 import org.agrona.MutableDirectBuffer;
+import org.agrona.collections.IntArrayList;
 import org.agrona.collections.Long2ObjectHashMap;
+import org.agrona.collections.LongArrayList;
 import org.agrona.concurrent.UnsafeBuffer;
-import org.reaktivity.nukleus.buffer.BufferPool;
+import org.reaktivity.nukleus.buffer.DirectBufferBuilder;
+import org.reaktivity.nukleus.buffer.MemoryManager;
 import org.reaktivity.nukleus.function.MessageConsumer;
 import org.reaktivity.nukleus.function.MessageFunction;
 import org.reaktivity.nukleus.function.MessagePredicate;
 import org.reaktivity.nukleus.route.RouteManager;
 import org.reaktivity.nukleus.stream.StreamFactory;
+import org.reaktivity.nukleus.tls.internal.FrameFlags;
 import org.reaktivity.nukleus.tls.internal.TlsConfiguration;
+import org.reaktivity.nukleus.tls.internal.stream.util.AckedRegionBuilder;
 import org.reaktivity.nukleus.tls.internal.types.Flyweight;
-import org.reaktivity.nukleus.tls.internal.types.OctetsFW;
+import org.reaktivity.nukleus.tls.internal.types.ListFW;
+import org.reaktivity.nukleus.tls.internal.types.ListFW.Builder;
 import org.reaktivity.nukleus.tls.internal.types.control.RouteFW;
 import org.reaktivity.nukleus.tls.internal.types.control.TlsRouteExFW;
-import org.reaktivity.nukleus.tls.internal.types.stream.AbortFW;
+import org.reaktivity.nukleus.tls.internal.types.stream.AckFW;
 import org.reaktivity.nukleus.tls.internal.types.stream.BeginFW;
-import org.reaktivity.nukleus.tls.internal.types.stream.DataFW;
-import org.reaktivity.nukleus.tls.internal.types.stream.EndFW;
-import org.reaktivity.nukleus.tls.internal.types.stream.ResetFW;
+import org.reaktivity.nukleus.tls.internal.types.stream.RegionFW;
 import org.reaktivity.nukleus.tls.internal.types.stream.TlsBeginExFW;
-import org.reaktivity.nukleus.tls.internal.types.stream.WindowFW;
+import org.reaktivity.nukleus.tls.internal.types.stream.TransferFW;
+import org.reaktivity.reaktor.internal.buffer.DefaultDirectBufferBuilder;
 
 public final class ServerStreamFactory implements StreamFactory
 {
+    private static final ListFW<RegionFW> EMPTY_REGION_RO;
```

Comment:
`EMPTY_REGIONS` ?

Reply:
The networkReplyMemoryManager (maybe needs a better name) takes bytesConsumed, bytesProduced, network outbuffer, regions used to generate the bytes, and a region builder.

It is used in doTransfer after encrypting the regions.  The regions used to create the bytes are hidden in the memory near the offset.

This static empty is used to pass into that method when no regions are used to generate the input (We can add another method with empty instead?)